### PR TITLE
Add WAF rules to protect Jenkins endpoint

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
Adding one more layer of security by adding WAF rules and associating with the load balancer. This prevents DDoS attack when endpoint is public.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by prudhvigodithi — APPROVED** [review]

File: lib/ci-stack.ts [hunk 1]
```diff
@@ -202,6 +203,10 @@ export class CIStack extends Stack {
       accessLogBucket: auditloggingS3Bucket.bucket,
     });
 
+    const waf = new JenkinsWAF(this, {
+      loadBalancer: externalLoadBalancer.loadBalancer,
```
Comment (peterzhuamazon) [inline lib/ci-stack.ts:207]: So this pointed to the Jenkins ALB? And will dynamically adjust if the ALB get replaced?

Comment (gaiksaya) [reply lib/ci-stack.ts:207]: Yes!

File: lib/security/waf.ts [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+];
+
+export class WAF extends CfnWebACL {
+  constructor(scope: Construct, id: string) {
+    super(scope, id, {
+      defaultAction: { allow: {} },
+      visibilityConfig: {
+        cloudWatchMetricsEnabled: true,
+        metricName: 'jenkins-WAF',
+        sampledRequestsEnabled: false,
```
Comment (peterzhuamazon) [inline lib/security/waf.ts:89]: Why is this one disabled here for sampled request?

**Review by peterzhuamazon — APPROVED** [review]

File: lib/security/waf.ts (hunk 2 above)
Comment (gaiksaya) [reply lib/security/waf.ts]: Good catch. Enabled it now. Please check

**Comment by prudhvigodithi** [comment]
Also this `waf.node.addDependency(externalLoadBalancer)` should be added to avoid `waf` creating before the `externalLoadBalancer`
```
export class JenkinsWAF extends Stack {
  constructor(scope: Construct, id: string, props: WafProps) {
    const waf = new WAF(this, 'WAFv2');
    // Create an association with the alb
    new WebACLAssociation(this, 'wafALBassociation', {
      resourceArn: props.loadBalancer.loadBalancerArn,
      webAclArn: waf.attrArn,
    });
  }
}
```
Then 
```
    const waf = new JenkinsWAF(this, {
      loadBalancer: externalLoadBalancer.loadBalancer,
    });
    waf.node.addDependency(externalLoadBalancer)
```

**Comment by gaiksaya** [comment]
CDK is smart enough to detect that @prudhvigodithi. It creates the dependency graph automatically for resources. Regarding stacks we need to specify.

**Pull request merged**